dask_expr/_shuffle.py

```diff
@@ -857,7 +857,9 @@ def _simplify_up(self, parent):
                 return NSmallest(self.frame, n=parent.n, _columns=self.by)
         if isinstance(parent, Filter):
             return SortValues(
-                Filter(self.frame, parent.predicate.substitute({self: self.frame})),
+                Filter(
+                    self.frame, parent.predicate.substitute({self._name: self.frame})
```

Comment:
This API seems unpleasant.  Maybe we should change the API to one of the following:

```python
def subs(self, old, new):

>>> self.subs(self, self.frame)
```

or 

```python
def subs(self, pairs):

>>> self.subs([(self, self.frame)])
```

Reply:
I'll suggest the former until we need the latter.  (SymPy, for reference, does both).

Reply:
You want to get rid of the dict?

And you also want to convert from an expression to the name inside of substitute? 

Just checking that I am not misunderstanding anything

Reply:
Right now this method requires users to use the `._name` API to use it effectively, which seems off.  I think that the API here should consume and produce expressions.  I wouldn't expect users to know that they have to use the `._name` attribute.  

Given that we can't put exprs in dicts as keys, we need to find some other way to craft the API I think.  

If we expect the common case to be a single 1-1 substitution, then putting the old and new as separate arguments should work

```python
x.subs({y: z}) # before
x.subs(y, z)   # after
```

Reply:
This seems better than 

```python
x.subs({y._name: z})
```

Reply:
Yep makes sense, updated